### DISCUSSION TITLE
usePrepareContractWrite Issue

### DISCUSSION THREAD

**Discussion by capslocklabs** [wevm/wagmi#3231] [Q&A]
can someone tell me why usePrepareContract is interacting with the contract when I am clicking on the rainbowkit connect button and getting the error "You do not have enough ethers..." however i do have ethers in my wallet?

i have installed rainbow kit and trying to use wagmi to achieve the mint functionality. 
------------here is my index.tsx-------------------
`import { ConnectButton } from '@rainbow-me/rainbowkit';
import type { NextPage } from 'next';
//import Head from 'next/head';
import styles from '../styles/Home.module.css';
import { abi } from '../contract-abi';
import { usePrepareContractWrite, useContractWrite } from 'wagmi';

const Home: NextPage = () => {
  const mintAmount = 1;  
  const { config } = usePrepareContractWrite({ address: '0x5210ED1328104a2AF6e6D53d396f8CdDAf68Ar68', abi: abi, functionName: 'mint', args: [mintAmount]});
  const { data, isLoading, write } = useContractWrite(config);
  return (
    <div className={styles.container}>
        <ConnectButton />
        <button onClick={() => write}>
          {isLoading ? 'Minting...' : 'Mint'}
        </button>
     </div>
  );
}
export default Home;
`--------------------------here is my app.tsx---------------------
import '../styles/globals.css';
import '@rainbow-me/rainbowkit/styles.css';
import { getDefaultWallets, RainbowKitProvider } from '@rainbow-me/rainbowkit';
import type { AppProps } from 'next/app';
import { configureChains, createConfig, WagmiConfig } from 'wagmi';
import { arbitrum, goerli, mainnet, optimism, polygon, base, zora } from 'wagmi/chains';
import { publicProvider } from 'wagmi/providers/public';

const { chains, publicClient, webSocketPublicClient } = configureChains(
  [
    goerli,
    mainnet,
    polygon,
    optimism,
    arbitrum,
    base,
    zora,
    ...(process.env.NEXT_PUBLIC_ENABLE_TESTNETS === 'true' ? [goerli] : []),
  ],
  [publicProvider()]
);

const { connectors } = getDefaultWallets({
  appName: 'G Test',
  projectId: 'my-id-here',
  chains,
});

const wagmiConfig = createConfig({
  autoConnect: false,
  connectors,
  publicClient,
  webSocketPublicClient,
});

function MyApp({ Component, pageProps }: AppProps) {
  return (
    <WagmiConfig config={wagmiConfig}>
      <RainbowKitProvider chains={chains}>
        <Component {...pageProps} />
      </RainbowKitProvider>
    </WagmiConfig>
  );
}

export default MyApp;

**Comment by capslocklabs** [comment]
can any one help?

**Comment by POKENA7** [comment]
@capslocklabs 
If you need to pay ETH for mint, need to add a value attribute.
https://wagmi.sh/react/prepare-hooks/usePrepareContractWrite#value-optional

Or,  it might be better to check the contract implementation.
When `mint()` is possibly reverted, an error occurs that there is not enough eth for gas.
